src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,164 @@
+     * @throws MissingParametersException When there are missing parameters for either main or sub commands.
+     * @throws InvalidParametersException When there are invalid parameters for sub commands.
+     * @throws InvalidDateFormatException When there are invalid date formats for sub commands.
+     * @throws MissingCommandException When sub command for main command is missing.
+     */
+    public static Command parse(String inputString) throws InvalidCommandException, MissingParametersException,
+            InvalidParametersException, InvalidDateFormatException, MissingCommandException {
+        // Split text into two.
+        // With index 0 be the first word of user input, and index 1 containing the rest of the string.
+        inputString = inputString.toLowerCase();
+        splitText = inputString.split(" ", 2);
+
+        String command = splitText[0];
+        String task;
+
+        switch (command) {
+        case "bye":
+            return new ExitCommand();
+        case "list":
+            return new ListCommand();
+        case "delete":
+            checkLength();
+            return new DeleteCommand(splitText[1]);
+        case "todo":
+            checkLength();
+            return new AddCommand(new ToDo(splitText[1]));
+        case "event":
+            checkLength();
+            LocalDate from = getDateWithCommand(splitText[1], "from");
+            LocalDate to = getDateWithCommand(splitText[1], "to");
+            task = getTask(splitText[1]);
+            return new AddCommand(new Event(task, from, to));
+        case "deadline":
+            checkLength();
+            LocalDate by = getDateWithCommand(splitText[1], "by");
+            task = getTask(splitText[1]);
+            return new AddCommand(new Deadline(task, by));
+        case "mark":
+            checkLength();
+            return new ChangeMarkCommand(splitText[1], true);
+        case "unmark":
+            checkLength();
+            return new ChangeMarkCommand(splitText[1], false);
+        case "find":
+            checkLength();
+            return new FindCommand(splitText[1]);
+        default:
+            throw new InvalidCommandException("I don't understand.");
+        }
+    }
+
+    /**
+     * Gets LocalDate parameter based on String sub commands.
+     *
+     * @param str String to search in.
+     * @param command  Sub Command to search for.
+     * @return LocalDate parameter based on Sub Command.
+     * @throws MissingCommandException When Sub Command cannot be found.
+     * @throws MissingParametersException When there are no parameters for Sub Command.
+     * @throws InvalidDateFormatException When the parameter does not have a proper LocalDate format.
+     */
+    private static LocalDate getDateWithCommand(String str, String command) throws MissingCommandException,
+            MissingParametersException, InvalidDateFormatException {
+        boolean found = false;
```

Comment:
Maybe the name of the boolean can be isFound?

Reply:
Yup done from previous comment, thanks!